index.js

```diff
@@ -144,13 +146,288 @@ function hash(msgVal) {
   return '%' + ssbKeys.hash(encode(msgVal)).replace('.sha256', '.bbmsg-v1')
 }
 
-// FIXME: might split this out and add validateBatch
-function validateSingle(bmsg, previous) {}
+/**
+ * Validate a decoded bendy-butt message value.
+ *
+ * @param {Object} msgVal - an object compatible with ssb/classic `msg.value`
+ * @param {Object} previousMsg - a decoded `msgVal` object
+ * @param {Buffer | string | null} hmacKey - a valid hmac key for signature verification
+ * @returns {Object | string} an `Error` object with descriptive message or success string
```

Comment:
Returning a 'message is valid' in the successful case is a bit odd, it's unlike the other validation return types we've made so far. I am still not sure what is the best way (to return or to throw?) but it's helpful to think of "validate" as "findErrors" and if no errors are found, something "empty" is returned, maybe null or undefined or false.

So in this case I suggest changing the return type to `Object | undefined`. This would actually bring `validateSingle` in line with `validateSignature`, `validatePrevious`, `validateTypeFormatData`, `validateHmacKey` which also return undefined on success. Then, README and tests need to be updated too.

Reply:
Totally agree with this! I just noticed the inconsistency while working on meta feed validation. `Object | undefined` return type seems like the way to go. Will make the changes now.